dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Tearing.java

```diff
@@ -0,0 +1,308 @@
+        if (logger.isDebugEnabled()) {
+            final List<SourceLocation> sortedLocs = map.keySet().stream()
+                    .map(e -> e.getMetadata(SourceLocation.class))
+                    .filter(Objects::nonNull)
+                    .distinct()
+                    .sorted(Ordering.usingToString())
+                    .toList();
+            StringBuilder info = new StringBuilder();
+            info.append("\n").append("======== Tearing source information ========");
+            sortedLocs.forEach(loc -> info.append("\n\t").append(loc));
+            info.append("\n").append("============================================");
+            logger.debug(info);
+        }
+        IdReassignment.newInstance().run(program);
+    }
+
+    private int tearInits(Program program, AliasAnalysis alias, boolean bigEndian) {
+        int numTearings = 0;
+        for (Init init : program.getThreadEvents(Init.class)) {
+            final List<Integer> offsets = alias.mayMixedSizeAccesses(init);
+            if (offsets.isEmpty()) {
+                continue;
+            }
+            final int bytes = checkBytes(init, offsets);
+            final MemoryObject base = init.getBase();
+            final int initOffset = init.getOffset();
+            final Expression value = init.getValue();
+            // Tear initial values
+            final int frontBegin = bigEndian ? bytes - offsets.get(0) : 0;
+            final int frontEnd = bigEndian ? bytes : offsets.get(0);
+            final Expression frontValue = expressions.makeIntExtract(value, 8 * frontBegin, 8 * frontEnd - 1);
+            base.setInitialValue(initOffset, frontValue);
+            for (int i = 0; i < offsets.size(); i++) {
+                final int offset = offsets.get(i);
+                final int next = i + 1 < offsets.size() ? offsets.get(i + 1) : bytes;
+                final int begin = bigEndian ? bytes - next : offset;
+                final int end = bigEndian ? bytes - offset : next;
+                final Expression tearedValue = expressions.makeIntExtract(value, 8 * begin, 8 * end - 1);
+                base.setInitialValue(initOffset + offset, tearedValue);
+            }
+            // Tear init event
+            init.setMemValue(frontValue);
+            init.setAccessType(frontValue.getType());
+            for (int begin : offsets) {
+                program.addInit(base, initOffset + begin);
+            }
+            numTearings++;
+        }
+        tearExpressions(program);
+        return numTearings;
+    }
+
+    private void tearExpressions(Program program) {
+        //TODO currently, FinalMemoryValue only occurs in the program's final state expressions.
+        final Expression specification = program.getSpecification();
+        final Expression filter = program.getFilterSpecification();
+        if (specification == null && filter == null) {
+            return;
+        }
```

Comment:
I changed `filterSpec` to never be null, but simply `true` if there is no filter. So the null checks for that one can go.